### ISSUE TITLE
Function for `timeWindow`

### ISSUE THREAD

**Issue by AlexRMU** [fastify/fastify-rate-limit#283]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Add a function so that you can do a limit, for example 10 per minute AND 100 per hour AND 1000 per day.

### Example

```js
rateLimit: {
    max: async (req, key) => { /*... */ },
    timeWindow: async (req, key) => { /*... */ },
    // or
    max: {
        // key = window
        // value = max
        [60 * 1000]: 10,
        [60 * 60 * 1000]: async (req, key) => { /*... */ },
    }
}
```

**Comment by Uzlopak** [comment]
Well this feature request raises questions. What are the headers on the response you gonna set? I think it is useful what you suggest, but needs more specs. So please elaborate.

**Comment by mindrunner** [comment]
+1 for this feature :)

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by AlexRMU** [comment]
I tried, but realized that it would take a lot of architecture changes.
I can help with tips or a review.

**Comment by gurgunday** [comment]
I think something similar can be achieved by using a function for the max parameter

**Comment by mindrunner** [comment]
https://github.com/fastify/fastify-rate-limit/pull/357

**Comment by mindrunner** [comment]
I added `function` feature for `timeWindow`. Please have a look  :)

**Comment by gurgunday** [comment]
Closed by #357
